scripts/src/javatests/org/oppia/android/scripts/ci/ComputeChangedFilesTest.kt

```diff
@@ -0,0 +1,826 @@
+    createAndCommitFile("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+
+    val reportedFiles = runScript()
+
+    assertThat(reportedFiles).isEmpty()
+  }
+
+  @Test
+  fun testUtility_featureBranch_noChanges_computeAllFiles_returnsAllFiles() {
+    initializeEmptyGitRepository()
+    createFiles("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+
+    val reportedFiles = runScript(computeAllFiles = true)
+
+    // Even though there are no changes, all files should be returned since that was requested via
+    // a command argument.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList)
+      .containsExactly("app/First.kt", "app/Second.kt", "app/Third.kt")
+  }
+
+  @Test
+  fun testUtility_featureBranch_fileChange_committed_returnsChangedFile() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+    changeAndCommitFile("First", subPackage = "app")
+
+    val reportedFiles = runScript()
+
+    // Only the first file should be reported since the file itself was changed & committed.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).containsExactly("app/First.kt")
+  }
+
+  @Test
+  fun testUtility_featureBranch_fileChange_staged_returnsChangedFile() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+    changeAndStageFile("First", subPackage = "app")
+
+    val reportedFiles = runScript()
+
+    // Only the first file should be reported since the file itself was changed & staged.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).containsExactly("app/First.kt")
+  }
+
+  @Test
+  fun testUtility_featureBranch_fileChange_unstaged_returnsChangedFile() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+    changeAndCommitFile("First", subPackage = "app")
+
+    val reportedFiles = runScript()
+
+    // Only the first file should be reported since the file itself was changed & staged.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).containsExactly("app/First.kt")
+  }
+
+  @Test
+  fun testUtility_featureBranch_newFile_untracked_returnsChangedFile() {
+    initializeEmptyGitRepository()
+    createAndCommitFile("First", "Second", "Third", subPackage = "app")
+    switchToFeatureBranch()
+    createFiles("NewUntrackedFile", subPackage = "data")
+
+    val reportedFiles = runScript()
+
+    // Only the first file should be reported since the file itself was changed & staged.
+    assertThat(reportedFiles).hasSize(1)
+    assertThat(reportedFiles.first().changedFilesList).containsExactly("data/NewUntrackedFile.kt")
```

Comment:
This seems backwards from what the test name is suggesting.

I worry a bit about overly matching ComputeAffectedTests--a lot of those tests are verifying that the script always only considers fully checked-in files for deltas (since it's meant to run in CI). If you plan to replicate that behavior, this test seems to be behaving wrong since untracked files should not be part of the 'changed files' result.

Reply:
Oh! Thank you so much for clarifying that, I was wondering if my wsl and git are functioning wrong as it was printing out a bunch of other files (while ci didn't), seems as you said I was doing it all wrong. And also, I only had a very high-level understanding of computing affected test targets and tried replicating it.

So, now I changed the script implementation to acquire **only the committed files** by using the existing method,

```
private fun retrieveChangedCommittedFiles(): List<String> {
    return executeGitCommand("diff --name-only ${computeCommitRange()}")
  }
```

hope now the script should be functioning in the way it should if not will adjust it as per feedback asap.

---

with this will also modify the tests and update here.